packages/narwallets/src/lib/narwallets.ts

```diff
@@ -0,0 +1,274 @@
+import { isMobile } from "is-mobile";
+import {
+  WalletModuleFactory,
+  WalletBehaviourFactory,
+  InjectedWallet,
+  Account,
+  Action,
+} from "@near-wallet-selector/core";
+import { providers } from "near-api-js";
+
+export interface SignAndSendTransactionParams {
+  signerId?: string;
+  receiverId?: string;
+  actions: Array<Action>;
+}
+export interface NarwalletsParams {
+  iconUrl?: string;
+}
+
+interface PendingPromises {
+  id: number;
+  code: string;
+  // Note: the result from different action may give different results, like boolean or FinalExecutionOutcome. Since the response may be many things, we're leaving any here
+  resolve: (value: any) => void;
+  reject: (reason?: any) => void;
+  timeout?: NodeJS.Timeout;
+}
+
+const NARWALLETS_CODES = {
+  SIGN_IN: "sign-in",
+  IS_INSTALLED: "is-installed",
+  IS_SIGNED_IN: "is-signed-in",
+  SIGN_OUT: "sign-out",
+  GET_ACCOUNT_ID: "get-account-id",
+  SIGN_AND_SEND_TRANSACTION: "sign-and-send-transaction",
+  SIGN_AND_SEND_TRANSACTIONS: "sign-and-send-transactions" 
+}
+
+let id = 0;
+const pendingPromises: Array<PendingPromises> = [];
+
+const isInstalled = async (): Promise<boolean> => {
+  // Note: sendToNarwallets throws if not installed
+  try {
+    await sendToNarwallets(NARWALLETS_CODES.IS_INSTALLED, true);
+  }
+  catch (ex) {
+    return false
+  }
+  return true;
+};
+
+const isSignedIn = (): Promise<boolean> => {
+  return sendToNarwallets(NARWALLETS_CODES.IS_SIGNED_IN, true);
+};
+
+const getAccountId = (): Promise<string> => {
+  return sendToNarwallets(NARWALLETS_CODES.GET_ACCOUNT_ID, true);
+};
+
+const callSignAndSendTransaction = (
+  params: SignAndSendTransactionParams
+): Promise<providers.FinalExecutionOutcome> => {
+  return sendToNarwallets(NARWALLETS_CODES.SIGN_AND_SEND_TRANSACTION, false, params);
+};
+
+const callSignAndSendTransactions = (
+  params: Array<SignAndSendTransactionParams>
+): Promise<
+  Array<providers.FinalExecutionOutcome>
+> => {
+  return sendToNarwallets(NARWALLETS_CODES.SIGN_AND_SEND_TRANSACTIONS, false, params);
+};
+
+const sendToNarwallets = (
+  code: string,
+  withTimeout = false,
+  params?: any
+): Promise<any> => {
+  const promise = new Promise<any>((resolve, reject) => {
+    id++;
+    let promiseTimeout;
+    if (withTimeout) {
+      promiseTimeout = setTimeout(() => {
+        return reject(Error("timeout"));
+      }, 2000);
+    }
+    pendingPromises.push({
+      id,
+      code,
+      resolve,
+      reject,
+      timeout: promiseTimeout,
+    });
+
+    window.postMessage({
+      id,
+      src: "ws",
+      type: "nw",
+      code,
+      dest: "ext",
+      params,
+    });
+  });
+  return promise;
+};
+
+function findPendingPromiseById(id: number): PendingPromises | undefined {
+  return pendingPromises.filter((c) => c.id == id)[0];
+}
```

Comment:
With ES6 arrow function you mean to remove the parenthesis from the (c)? 

Changed `id` to `id_wallet_selector`

Reply:
I mean this function should be something like this:

```ts
const findPendingPromiseById = (promiseId: number): PendingPromises | undefined => {
  return pendingPromises.filter((c) => c.id == promiseId)[0];
}
```

instead of:

```ts
function findPendingPromiseById(id: number): PendingPromises | undefined {
  return pendingPromises.filter((c) => c.id == id)[0];
}
```

Reply:
Done